test/unit/org/apache/cassandra/locator/SimpleStrategyTest.java

```diff
@@ -316,8 +320,37 @@ public void testSimpleStrategyThrowsConfigurationException() throws Configuratio
 
         Map<String, String> configOptions = new HashMap<>();
 
+        @SuppressWarnings("unused")
         SimpleStrategy strategy = new SimpleStrategy("ks", metadata, snitch, configOptions);
     }
+    
+    @Test
+    public void shouldReturnNoEndpointsForEmptyRing()
+    {
+        TokenMetadata metadata = new TokenMetadata();
+        
+        HashMap<String, String> configOptions = new HashMap<>();
+        configOptions.put("replication_factor", "1");
+        
+        SimpleStrategy strategy = new SimpleStrategy("ks", metadata, new SimpleSnitch(), configOptions);
+
+        EndpointsForRange replicas = strategy.calculateNaturalReplicas(null, metadata);
+        assertTrue(replicas.endpoints().isEmpty());
+    }
+
+    @Test
+    public void shouldWarnOnHigherReplicationFactorThanNodes()
```

Comment:
There is already coverage for this in [`AlterTest#testCreateAlterKeyspacesRFWarnings`](https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java#L279-L329), but it's nice to have it too. We could add an analogous test for `NetworkTopologyStrategyTest`:
```java
@Test
public void shouldWarnOnHigherReplicationFactorThanNodes()
{
    IEndpointSnitch snitch = new SimpleSnitch();

    Map<String, String> configOptions = new HashMap<>();
    configOptions.put(snitch.getDatacenter((InetAddressAndPort) null), "2");

    NetworkTopologyStrategy strategy = new NetworkTopologyStrategy("ks", new TokenMetadata(), snitch, configOptions);

    StorageService.instance.getTokenMetadata().updateHostId(UUID.randomUUID(), FBUtilities.getBroadcastAddressAndPort());
    ClientWarn.instance.captureWarnings();
    strategy.maybeWarnOnOptions();
    assertTrue(ClientWarn.instance.getWarnings().stream().anyMatch(s -> s.contains("Your replication factor")));
}
```